<?xml version="1.0"?>


<launch>
  <arg name="base_frame" default="base_footprint"/>
  <arg name="fixed_frame" default="map"/>
  <arg name="bot_params" default="turtlebot_params" />  
  <arg name="map" default="maze.yaml" />

    <!-- Shared controlTrajectory Generator -->
  <node pkg="trajectory_generator" type="trajectory_generator" name="trajectory_generator" output="screen">
    <rosparam command="load" file="$(find trajectory_generator)/config/settings.yaml" />
    <!--<remap from="/trajectory_generator/goal" to="/move_base_simple/goal"/>    
    <remap from="/trajectory_generator/trajectory" to="/global_planner_plugin/trajectory"/>--> 
    <!--remap from="/user/cmd_vel" to="/falcon/cmd_vel"/-->    
    <param name="node_active" value="true"/>
    <param name="publish_frequency" value="4"/>
  </node>

    <!-- hardware drivers, alternatively, use the 'bare' simulation launch files mbot/scatwheel_navigation -->
    <!-- <include file="$(find scat_launch)/launch/system.launch">
    </include> -->

    <!-- Launch Localization pipeline -->
    <include file="$(find scat_launch)/launch/localization.launch">
	  <arg name="map" value="$(arg map)" />
    </include>

    <!-- Joystick controllers. Use either joystick_filter or control_joystick (no filter) -->
  <node pkg="joystick_filter" type="joystick_filter" name="joystick_filter" output="screen">
    <rosparam command="load" file="$(find joystick_filter)/config/settings.yaml" />
    <remap from="/user/joy" to ="/joy_data/remote" />
    <param name="kv" value="0.5"/>
    <param name="kw" value="1.5"/>
    <param name="use_medianfilter" value="false"/>
    <param name="use_kalmannfilter" value="false"/>
  </node>

    <!-- Velocity filter. Filters final velocity command -->
  <node pkg="vel_safety_filter" type="vel_safety_filter" name="vel_safety_filter" output="screen">
    <rosparam command="load" file="$(find vel_safety_filter)/config/settings.yaml" />    
    <rosparam command="load" file="$(find scat_launch)/config/$(arg bot_params).yaml" />
    <remap from="/odom" to="/ekf/odom" />
    <param name="use_vel_filter" value="false"/>
    <param name="use_col_filter" value="false"/>
  </node>

    <!-- move_base node -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen" clear_params="true">
    <remap from="/odom" to="/ekf/odom"/>
    <remap from="/cmd_vel" to="/control/cmd_vel"/>

    <param name="base_global_planner" value="global_planner_plugin/GlobalPlannerPlugin" />    
    <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />

    <rosparam file="$(find scat_launch)/config/move_base_params.yaml" command="load" />
    <rosparam file="$(find scat_launch)/config/local_planner_params.yaml" command="load" />
    <rosparam file="$(find scat_launch)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find scat_launch)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find scat_launch)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find scat_launch)/config/global_costmap_params.yaml" command="load" />
    <param name="planner_frequency" value="0.5" />
  </node>

</launch>
